core/io/file_access_extension.cpp

```diff
@@ -0,0 +1,355 @@
+/**************************************************************************/
+/*  file_access_extension.cpp                                             */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "file_access_extension.h"
+
+#include "core/object/object.h"
+
+Error FileAccessExtension::open_internal(const String &p_path, int p_mode_flags) {
+	Error err = OK;
+	GDVIRTUAL_CALL(_open_internal, p_path, p_mode_flags, err);
+	return err;
+}
+
+uint64_t FileAccessExtension::_get_modified_time(const String &p_file) {
+	uint64_t time = 0;
+	GDVIRTUAL_CALL(__get_modified_time, p_file, time);
+	return time;
+}
+
+bool FileAccessExtension::is_open() const {
+	bool is_open = false;
+	GDVIRTUAL_CALL(_is_open, is_open);
+	return is_open;
+}
+
+String FileAccessExtension::get_path() const {
+	String path = "";
```

Comment:
```suggestion
	String path;
```
Unnecessary defaults (all non simple types)

Reply:
![image](https://github.com/godotengine/godot/assets/270928/ef12b569-76f1-4f0e-9488-b134bd7e45e7)

The compiler is screaming at me. I'll initialize the values.

Reply:
`String` is default initialized, so the suggestion is correct.

Primitive types lime `bool`, `int`, or enums the like `Error`, do need an explicit initialization.

Reply:
Oh, I understand better now @AThousandShips's comment. Thanks @akien-mga!